### DISCUSSION TITLE
Resolve binding clashes across multiple input maps

### DISCUSSION THREAD

**Discussion by ndarilek** [Leafwing-Studios/leafwing-input-manager#217] [Q&A]
Hey folks,

I have various plugins I've created to use across games where I expect to need the same behavior. So for instance, I have this in my `NavigationPlugin`:

```
#[derive(Actionlike, PartialEq, Eq, Clone, Copy, Hash, Debug)]
pub enum NavigationAction {
    Move,
    Rotate,
    SnapLeft,
    SnapRight,
    SnapCardinal,
    SnapReverse,
    Sprint,
}
```

Then, because I want to bind multiple keys to a single `VirtualDPad` input, I have actions for `Forward`, `Backward`, `Left` and `Right` alongside my game-specific actions. I believe this pattern, exporting and binding plugin-specific actions, was suggested as a best practice.

The problem arises when I bind, say, `Down` to my `Backward` action in my main actions mapping, but then also bind ctrl-down arrow to `NavigationAction::SnapReverse`. Clash detection doesn't work across separate `InputMap` instances and they're both triggered simultaneously.

I can see a few possible paths forward. Maybe there could be a `CompositeInputMap` composed of several submaps, or clash detection might be set up to take multiple `InputMap`s into account. I don't have a preference either way.

Or am I missing some additional step? Right now I'm adding several `InputMap`s to my player, which mostly works save for clash detection and a few other gamepad-related issues I'm currently debugging and may post about later today.

Thanks a bunch.

**Comment by alice-i-cecile** [comment]
Hmm. Can you make an issue out of this please? I think it's a flaw in the design.

**Reply by ndarilek** [reply]
Sure, should I just make an issue out of this discussion? I don't know if that does something meaningful or if it's more noisy.

FWIW, I'm using the latest dev branch and am getting lots of "Real clash" messages printed in my console. Not sure if that gives any indication as to what might be happening in the various discussions I've opened here today. :)

**Reply by alice-i-cecile** [reply]
I'm the maintainer; that would be useful to help my track work :)

As for the spam, use cargo update. I just released a patch!

**Reply by ndarilek** [reply]
OK, sorry, to be clear I wasn't sure if I should convert the discussion to an issue or open up an entirely new one. I haven't used Discussions before. :) I'll just convert it to an issue and we can go from there if the conversion is wonky.

Thanks for all your help so far.
